### ISSUE TITLE
Project Definition | Follow-up details

### ISSUE THREAD

**Issue by mateoclarke** [cityofaustin/atd-data-tech#4544]
I didn't dig too much into the causes of any of this yet but, I encountered a few buggy things about the form UI that I think we'll still need to address in the next pull request. 

- [ ] ~The text input (ex: project name) seemed a little laggy. Wondering if there's a more performant way to handle state changes.~
- [x] The Add Staff icon needs some styling and padding.
- [x] When I select a staff, and then remove them, the form hangs and gives me `TypeError: Cannot read property 'workgroup' of null` related to the `handleGroupChange` function.
- [x] Submit form returns GraphQL error.

More items from 12/9 discussion:
another small thing:
- [x] Workgroup doesn't update until you start typing in Notes @mddilley 
- [x] If you add another staff member, the new data for second staff members goes into the DB, but it looks the same (ex: notes). @mddilley 
    - Notes state needs to be repopulate on changes
- [x] switch the order of workgroup (move to the left) and role (move to the right)

**Comment by amenity** [comment]
@lazavazav Can you add an estimate to this? I think it makes sense to do before #3671 if you haven't started that yet.

**Comment by lazavazav** [comment]
The text input (ex: project name) seemed a little laggy. Wondering if there's a more performant way to handle state changes. (looking into it)
The Add Staff icon needs some styling and padding.  (Done)
When I select a staff, and then remove them, the form hangs and gives me TypeError: Cannot read property 'workgroup' of null related to the handleGroupChange function.  (still lookin into this)
Submit form returns GraphQL error. (Fixed this on branch #4507_lz_projNotes: need to confirm with Sergio if those changes are ok)

**Comment by amenity** [comment]
@lazavazav - will you add an estimate to this, please? And note if there are any PRs this is connected to. 🙏 😄

**Comment by johnclary** [comment]
@mateoclarke i had promised to split this into smaller issues, but i'll let you do that if any of these items gets big enough.

**Comment by mateoclarke** [comment]
another small thing:
- Workgroup doesn't update until you start typing in Notes
- If you add another staff member, the new data for second staff members goes into the DB, but it looks the same (ex: notes). 
- Notes state needs to be repopulate on changes
- switch the order of workgroup (move to the left) and role (move to the right)

**Comment by mateoclarke** [comment]
@mddilley and I caught up on these issues this morning. We weren't able to reproduce one of the bugs I had mentioned, so we struck the one out.

And the more we're thinking about the workgroup/notes issues, we think they might all be pretty related and not distinct tasks to split up.

Reassign to Mike since he needs more on his plate for today.

**Comment by mddilley** [comment]
The root of some of the UI bugs was an issue with updates to the objects nested in the `staffRows` array not being recognized as changes by React. I created new references by mapping the array and updated the relevant object by index so they are now recognized by the shallow comparison. Managing state with nested data structures is always tricky!

The notes bug was coming from each successive row being controlled by the same piece of state. I resolved this by tying each new notes field to the corresponding object in the `staffRows` array. I still need to refactor how to get the workgroup of a user to add to state. Will continute next week.
